### PR TITLE
Feature auth v1

### PR DESCRIPTION
**Pull request by sailesh97**
@adithnaveen @IrshadMulani @ayushk96

### CODE REVIEW

File: src/controllers/auth-controller.js [hunk 1]
```diff
@@ -35,6 +36,12 @@ export const addCustomer = async (req, res) => {
         let returnedCustomer = await newCustomer.save();
         res.send(returnedCustomer);
     } catch (err) {
+        if(err.message){
+            res.send({
+                error: err._message,
```
Comment (adithnaveen) [inline src/controllers/auth-controller.js:31]:
```suggestion
                error: err.code,
```

File: src/models/customer-model.js [hunk 2]
```diff
@@ -1,27 +1,30 @@
...
-export const CustomerSchema = new Schema(
+const CustomerSchema = new Schema(
     {
         customerName: {
             type: String,
-            required: "Please enter your fullname."
+            required: "Please enter your fullname.",
+            unique: true
         },
         phone: {
```
Comment (adithnaveen) [inline src/models/customer-model.js]:
```suggestion
        phone: Number
```

would be better as there is on validation

File: src/models/customer-model.js [hunk 3]
```diff
@@ -1,27 +1,30 @@
...
         email: {
-            type: String
-            // unique: true,
+            type: String,
+            unique: true
             // required: true,
             // dropDups: true
         },
         address: {
             type: String
```
Comment (adithnaveen) [inline src/models/customer-model.js]: @sailesh97  pls refer to earlier comment

**Review by adithnaveen — CHANGES_REQUESTED** [review]
pls look a the comments

File: src/controllers/auth-controller.js (hunk 1 above)
Comment (sailesh97) [reply src/controllers/auth-controller.js:31]: This is the code for error handling for duplicate users with same email/aadhar no/ username. The error object I'm getting is like below:
`
{
  "errors": {
    "customerName": {
    },
    "email": {
    },
    "aadharNo": {
    }
  },
  "_message": "user validation failed",
  "name": "ValidationError",
  "message": "user validation failed: customerName: Error, expected `customerName` to be unique. Value: `Sailesh Patra`, email: Error, expected `email` to be unique. Value: `sailes@gmail.com`, aadharNo: Error, expected `aadharNo` to be unique. Value: `4553-9009-4621`"
}
`

There's no "code" key in the error object.
@adithnaveen

File: src/models/customer-model.js (hunk 2 above)
Comment (sailesh97) [reply src/models/customer-model.js]: I'll do the neccessary changes here

**Review by adithnaveen — APPROVED** [review]
merging to master

**Pull request merged**